src/thunderbots/software/ai/hl/stp/tactic/goalie_tactic.cpp

```diff
@@ -0,0 +1,125 @@
+#include "ai/hl/stp/tactic/goalie_tactic.h"
+
+#include "ai/hl/stp/action/chip_action.h"
+#include "ai/hl/stp/action/move_action.h"
+#include "ai/hl/stp/evaluation/calc_best_shot.h"
+#include "geom/point.h"
+#include "geom/ray.h"
+#include "geom/segment.h"
+#include "geom/util.h"
+#include "shared/constants.h"
+
+GoalieTactic::GoalieTactic(const Ball &ball, const Field &field,
+                           const Team &friendly_team, const Team &enemy_team)
+    : ball(ball),
+      field(field),
+      friendly_team(friendly_team),
+      enemy_team(enemy_team),
+      Tactic(true)
+{
+}
+
+std::string GoalieTactic::getName() const
+{
+    return "Goalie Tactic";
+}
+
+void GoalieTactic::updateParams(
+    const Ball &ball, const Field &field, const Team &friendly_team,
+    const Team &enemy_team, const std::optional<Evaluation::EnemyThreat> &enemy_threat)
+{
+    // Update the parameters stored by this Tactic
+    this->ball          = ball;
+    this->field         = field;
+    this->friendly_team = friendly_team;
+    this->enemy_team    = enemy_team;
+    this->enemy_threat  = enemy_threat;
+}
+
+double GoalieTactic::calculateRobotCost(const Robot &robot, const World &world)
```

Comment:
Will we have issues with this because the goalie chosen robot?

Reply:
Possibly but I'm going to deal with goalie assignment in a separate PR. We will do some kind of explicit assignment